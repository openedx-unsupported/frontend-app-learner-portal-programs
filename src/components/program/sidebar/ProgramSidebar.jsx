import React, { useContext } from 'react';
import { sendTrackEvent } from '@edx/frontend-platform/analytics';
import { AppContext } from '@edx/frontend-platform/react';
import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { SidebarBlock } from '../../layout';

import Links from './Links';

function ProgramSidebar() {
  const {
    pageContext: {
      programDocuments, externalProgramWebsite,
    },
  } = useContext(AppContext);

  return (
    <>
      {programDocuments && programDocuments.display && (
        <SidebarBlock title={programDocuments.header} className="mb-5">
          <Links
            id={programDocuments.header.toLowerCase().split(' ').join('-')}
            links={programDocuments.documents}
            label="program documents"
          />
        </SidebarBlock>
      )}
      {externalProgramWebsite && externalProgramWebsite.display && (
        <SidebarBlock title={externalProgramWebsite.header} className="mb-5">
          {/* eslint-disable-next-line react/no-danger */}
          <div dangerouslySetInnerHTML={{ __html: externalProgramWebsite.description }} />
          <p>
            <a
              href={externalProgramWebsite.link.url}
              target="_blank"
              rel="noopener noreferrer"
              onClick={() => {
                sendTrackEvent('edx.learner_portal.school_portal_link.clicked');
              }}
            >
              {externalProgramWebsite.link.display_text}
              <FontAwesomeIcon
                className="ml-2 text-info"
                icon={faExternalLinkAlt}
                size="sm"
                aria-hidden={false}
                aria-label="opens in a new window"
              />
            </a>
          </p>
        </SidebarBlock>
      )}
      <SidebarBlock title="Get Technical Support">
        <p>
          <a
            href="https://support.edx.org/hc/en-us/sections/360007968853-Master-s-student-help"
            target="_blank"
            rel="noopener noreferrer"
            onClick={() => {
              sendTrackEvent('edx.learner_portal.support_link.clicked');
            }}
          >
            Go to edX help center
            <FontAwesomeIcon
              className="ml-2 text-info"
              icon={faExternalLinkAlt}
              size="sm"
              aria-hidden={false}
              aria-label="opens in a new window"
            />
          </a>
        </p>
      </SidebarBlock>
    </>
  );
}

export default ProgramSidebar;
