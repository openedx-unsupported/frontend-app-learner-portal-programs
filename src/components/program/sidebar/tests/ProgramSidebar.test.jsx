/* eslint-disable react/jsx-no-constructed-context-values */
import React from 'react';
import renderer from 'react-test-renderer';
import { AppContext } from '@edx/frontend-platform/react';

import ProgramSidebar from '../ProgramSidebar';

describe('<ProgramSidebar />', () => {
  it('renders correctly', () => {
    const pageContext = {
      programDocuments: null,
      externalProgramWebsite: 'https://edx.org',
    };
    const tree = renderer
      .create((
        <AppContext.Provider value={{ pageContext }}>
          <ProgramSidebar />
        </AppContext.Provider>
      ))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with links', () => {
    const pageContext = {
      programDocuments: {
        header: 'Program Documents',
        display: true,
        documents: [
          {
            display_text: 'Test Document',
            document: '/test/document',
          },
          {
            display_text: 'Test Link',
            url: 'example.com/1',
          },
          {
            display_text: 'Test Document 2',
            document: '/test/document2',
          },
        ],
      },
      externalProgramWebsite: 'https://edx.org',
    };
    const tree = renderer.create((
      <AppContext.Provider value={{ pageContext }}>
        <ProgramSidebar />
      </AppContext.Provider>
    )).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
