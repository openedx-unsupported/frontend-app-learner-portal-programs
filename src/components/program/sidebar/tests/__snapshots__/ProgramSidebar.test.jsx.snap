// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgramSidebar /> renders correctly 1`] = `
<div>
  <h4
    className="mb-2"
  >
    Get Technical Support
  </h4>
  <p>
    <a
      href="https://support.edx.org/hc/en-us/sections/360007968853-Master-s-student-help"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      Go to edX help center
      <svg
        aria-hidden={false}
        aria-label="opens in a new window"
        className="svg-inline--fa fa-external-link-alt fa-w-16 fa-sm ml-2 text-info"
        data-icon="external-link-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </p>
</div>
`;

exports[`<ProgramSidebar /> renders correctly with links 1`] = `
Array [
  <div
    className="mb-5"
  >
    <h4
      className="mb-2"
    >
      Program Documents
    </h4>
    <nav
      aria-label="program documents"
    >
      <ul
        className="list-unstyled mb-2"
        id="program-documents"
      >
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-info"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="/test/document"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Test Document
          </a>
        </li>
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-info"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="example.com/1"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Test Link
          </a>
        </li>
        <li
          className="mb-1"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-file fa-w-12 mr-2 text-info"
            data-icon="file"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            href="/test/document2"
            onClick={[Function]}
            rel="noopener noreferrer"
            target="_blank"
          >
            Test Document 2
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  <div>
    <h4
      className="mb-2"
    >
      Get Technical Support
    </h4>
    <p>
      <a
        href="https://support.edx.org/hc/en-us/sections/360007968853-Master-s-student-help"
        onClick={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        Go to edX help center
        <svg
          aria-hidden={false}
          aria-label="opens in a new window"
          className="svg-inline--fa fa-external-link-alt fa-w-16 fa-sm ml-2 text-info"
          data-icon="external-link-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </a>
    </p>
  </div>,
]
`;
